pyanaconda/storage/root.py

```diff
@@ -124,28 +124,28 @@ def _find_existing_installations(devicetree):
             device.teardown()
             continue
 
+        architecture = None
+        product = None
+        version = None
+
         try:
             (architecture, product, version) = get_release_string(chroot=sysroot)
         except ValueError:
-            name = _("Linux on %s") % device.name
-        else:
-            # I'd like to make this finer grained, but it'd be very difficult
-            # to translate.
-            if not product or not version or not architecture:
-                name = _("Unknown Linux")
-            elif "linux" in product.lower():
-                name = _("%(product)s %(version)s for %(arch)s") % \
-                    {"product": product, "version": version, "arch": architecture}
-            else:
-                name = _("%(product)s Linux %(version)s for %(arch)s") % \
-                    {"product": product, "version": version, "arch": architecture}
+            pass
```

Comment:
I'm not quite happy with this solution. It's not really visible that you depend on the fact that `architecture`, `product`, `version` variables are `None` because they are not set. Could you please make this somehow more obvious?

Reply:
The function `get_release_string` can already return None values, so I have moved the try-except block there and simplified this code.

Reply:
Oh it can even return None. In that case it's even more confusing... :(